main.c

```diff
@@ -84,6 +84,8 @@ ioline_t ledRows[NUM_ROW * 4] = {
 
 #define KEY_COUNT                   70
 
+#define SERIAL_CHECK_FREQUENCY      0xFFF
```

Comment:
probably write this in decimal since it is a Hz value.

Reply:
I don't like this name either. I use it as frequency in most places but this https://github.com/OpenAnnePro/AnnePro2-Shine/blob/f21a57a100f3b3ee5998d3d483f1631fd0caa8c6/main.c#L437

Here, using bitwise operator instead of modulo since it is called often. Not sure if there is a better way to write this part 🤔

Reply:
so doing mod is perfectly fine. since the RHS is a constant. If it is a power of 2, the compiler will optimize. You can verify this by looking the the produced lst file. (Or just objdump it yourself)

Reply:
I made it simpler, by resetting the counter. All should be OK now.